File: C#/0111. Minimum Depth of Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int MinDepth(TreeNode root) {
        if (root==null){
            return 0;
        }
        
        int cnt=0;
        
        Queue<TreeNode> rootQueue=new Queue<TreeNode>();
        rootQueue.Enqueue(root);
        while(rootQueue.Count!=0){
            cnt+=1;
            int lenRootQueue=rootQueue.Count;
            while(lenRootQueue>0){
                lenRootQueue--;
                TreeNode curRoot=rootQueue.Dequeue();
                if(curRoot.left==null && curRoot.right==null){
                    return cnt;
                }
                if(curRoot.left!=null){
                    rootQueue.Enqueue(curRoot.left);
                }
                if(curRoot.right!=null){
                    rootQueue.Enqueue(curRoot.right);
                }
            }
        }
        return 0;
    }
}

File: 20200215Python-China/0111. Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if not root:
            return 0
        if root.left and root.right:
            return 1+min(self.minDepth(root.left),self.minDepth(root.right))
        elif root.left and not root.right:
            return 1+self.minDepth(root.left)
        elif not root.left and root.right:
            return 1+self.minDepth(root.right)
        else:
            return 1
